sources/web/datalab/settings.ts

```diff
@@ -118,8 +118,54 @@ function copyDefaultUserSettings(userId: string) {
   const defaultUserSettingsPath = path.join(__dirname, 'config', DEFAULT_USER_SETTINGS_FILE);
   console.log('Copying default settings: ' + defaultUserSettingsPath + ' to: ' + userSettingsPath);
   // Copy the default user settings file into user's directory.
-  const data = fs.readFileSync(defaultUserSettingsPath);
-  fs.writeFileSync(userSettingsPath, data);
+  const defaultUserSettings = fs.readFileSync(defaultUserSettingsPath, {encoding: 'utf8'});
+  const initialUserSettings = process.env.DATALAB_INITIAL_USER_SETTINGS;
+  const mergedUserSettings : string = initialUserSettings ?
+      mergeUserSettings(defaultUserSettings, initialUserSettings) : defaultUserSettings;
+  fs.writeFileSync(userSettingsPath, mergedUserSettings);
+  // writeFileSync does not return a status; let's see if it wrote a file.
+  if (!fs.existsSync(userSettingsPath)) {
+    console.log('Failed to write new user settings file ' + userSettingsPath);
+  }
+}
+
+/**
+ * Merges two sets of user settings, giving priority to the second.
+ * Exported for testing.
+ */
+export function mergeUserSettings(defaultUserSettings: string, initialUserSettings: string): string {
+  let parsedDefaultUserSettings;
+  try {
+    if (defaultUserSettings) {
```

Comment:
Instead of these five lines (same below), can we just do `JSON.parse(defaultUserSettings || '{}')`?